### PR TITLE
Temtechie recent dev

### PR DESCRIPTION
**Pull request by temtechie**
@abelbordonado @Greyastratos created a  PR.

I have dropped some message on slack. 
Please ignore some commented code, due to the new design changes

### CODE REVIEW

File: messages/de.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
     "marketplace": {
       "title": "Marktplatz",
       "market": "Markt",
-      "myPositions": "Meine Positionen"
```
Comment (abelbordonado) [inline messages/de.json:23]: We dont really need to keep deutch, can remove from now and in Spanish can just have english. 

When we are complete we will translate it at once. Otherwise will take time from you

File: src/app/[locale]/(routes)/collection/page.tsx [hunk 2]
```diff
@@ -1,61 +1,59 @@
...
+    if (isLoading) {
+      return <div>Fetching block number</div>;
+    }
+    if (isError) {
+      return <div>Error Fetching block number</div>;
+    }
+    let amountAdd = 216000 * n; //216000 is the number of blocks that represents 30 days
+    let loanRepayDeadLineAddedAmount = BigInt(amountAdd);
+    let loanRepayDeadLine = loanRepayDeadLineAddedAmount;
+    if (data?.toString().length) {
```
Comment (abelbordonado) [inline src/app/[locale]/(routes)/collection/page.tsx:29]: Lets not use `let` use`const` it self

When you are using let you are telling the reader the variable is going to be modified down the code again which makes hard to follow up

File: src/app/[locale]/(routes)/collection/page.tsx [hunk 3]
```diff
@@ -1,61 +1,59 @@
...
+
+  const handleGetBlockTime = (s: string) => {
+    const n = parseInt(s);
+    if (isLoading) {
+      return <div>Fetching block number</div>;
+    }
+    if (isError) {
+      return <div>Error Fetching block number</div>;
+    }
+    let amountAdd = 216000 * n; //216000 is the number of blocks that represents 30 days
```
Comment (abelbordonado) [inline src/app/[locale]/(routes)/collection/page.tsx:26]: 216000 is not a constant number needs to come from web3.

If is not part of the PR lets add a task to modify this

File: src/app/[locale]/(routes)/collection/page.tsx [hunk 4]
```diff
@@ -1,61 +1,59 @@
...
+  };
+
+  const { address } = useAccount();
+
+  const checkAddres = (address: Address | undefined) => {
+    if (address !== undefined && address.length) {
+      return <Balance address={address} />;
+    } else {
+      return <span>nope</span>;
     }
```
Comment (abelbordonado) [inline src/app/[locale]/(routes)/collection/page.tsx:44]: This else is not needed as the `if` uses `return`

File: src/app/[locale]/(routes)/collection/page.tsx [hunk 5]
```diff
@@ -1,61 +1,59 @@
...
   };
 
-   return ( <>
-    <Cardcontainer/>
-    {checkAddres(address)}
-    <label>
+  return (
+    <>
+      <Cardcontainer />
+      {checkAddres(address)}
```
Comment (abelbordonado) [inline src/app/[locale]/(routes)/collection/page.tsx:55]: Check address is rednering? Can we have it as component `<CheckAddress address={address} />

File: src/app/[locale]/(routes)/mypositions/page.tsx [hunk 6]
```diff
@@ -9,7 +11,7 @@ const page: FC<pageProps> = () => {
     <section className="my-4 xl:my-8 space-y-4 lg:space-y-8">
       <PageTitle title="My Positions" />
       <PositionTabs />
-      <PositionCards />
+      <PositionCards positionCardsData={positionCardsData} />
```
Comment (abelbordonado) [inline src/app/[locale]/(routes)/mypositions/page.tsx:14]: This data comes from an inpirt.

I belive is going to be replace later on.
But the import could have being just done inside

File: src/app/[locale]/layout.tsx [hunk 7]
```diff
@@ -1,79 +1,59 @@
...
-				</ReduxProvider>
-			</body>
-		</html>
-	);
+  const messages = await getMessages(locale);
+
+  return (
+    <html lang={locale}>
+      <body suppressHydrationWarning={true}>
+        <ReduxProvider>
```
Comment (abelbordonado) [inline src/app/[locale]/layout.tsx:45]: Good that you use the prettier to fix this page 👍

File: src/app/component/login/Login.tsx [hunk 8]
```diff
@@ -40,88 +49,134 @@ const Login = () => {
...
+                  <button
+                    key={connector.id}
+                    onClick={() => connect({ connector })}
+                    className="flex cursor-pointer items-center w-full mb-2 bg-white text-black rounded-lg flex-row  hover:bg-gray-100"
+                  >
+                    <div className="rounded-l-lg px-3 py-2 bg-appColor1">
+                      {connector.name === "MetaMask" && <MetaMask />}
+                      {connector.name === "Coinbase Wallet" && <CoinBase />}
+                      {connector.name === "WalletConnect" && <WalletConnect />}
+                      {connector.name === "Injected" && <WalletConnect />}
```
Comment (abelbordonado) [inline src/app/component/login/Login.tsx:159]: This contenct is duplicated in other file down here. 
Can we create something common ? (check how much must be common and how much not)

File: src/hooks/useAuth.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
+'use client'
+import { useState } from 'react';
+import { useAccount, useDisconnect } from 'wagmi';
+
+const useAuth = () => {
```
Comment (abelbordonado) [inline src/hooks/useAuth.ts:5]: Thanks, I was asking for this so long

File: messages/de.json (hunk 1 above)
Comment (temtechie) [reply messages/de.json:23]: Okay let me just keep it , going further I will not be updating it anymore

File: src/app/[locale]/(routes)/collection/page.tsx (hunk 2 above)
Comment (temtechie) [reply src/app/[locale]/(routes)/collection/page.tsx:29]: this code was written by @Greyastratos ,
But I will make the amendment

File: src/app/[locale]/(routes)/collection/page.tsx (hunk 3 above)
Comment (temtechie) [reply src/app/[locale]/(routes)/collection/page.tsx:26]: please add a task to modify it then 

//216000 is the number of blocks that represents 30 days

File: src/app/[locale]/(routes)/collection/page.tsx (hunk 4 above)
Comment (temtechie) [reply src/app/[locale]/(routes)/collection/page.tsx:44]: I have updated the function to the code below 

`  const checkAddres = (address: Address | undefined) => {
    if (address !== undefined && address.length) {
      console.log(address);
      return <Balance address={address} />;
    }

    return <span>nope</span>;
  };`

File: src/app/[locale]/(routes)/collection/page.tsx (hunk 5 above)
Comment (temtechie) [reply src/app/[locale]/(routes)/collection/page.tsx:55]: @Greyastratos  will handle it .

File: src/app/[locale]/(routes)/mypositions/page.tsx (hunk 6 above)
Comment (temtechie) [reply src/app/[locale]/(routes)/mypositions/page.tsx:14]: I just made it to be passed as prop, even if its coming from input we still need to pass the data to the component. the component is a reusable component

File: src/app/component/login/Login.tsx (hunk 8 above)
Comment (temtechie) [reply src/app/component/login/Login.tsx:159]: this Login will be removed. its not needed anymore. 

the content will be used one place after the login component is deleted. I will handle that later
